org.elasticsearch.snapshots.SnapshotResiliencyTests.testConcurrentSnapshotDeleteAndDeleteIndex failure

**Build scan**:
https://gradle-enterprise.elastic.co/s/vpvjriaf3z6r2


**Repro line**:
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.snapshots.SnapshotResiliencyTests.testConcurrentSnapshotDeleteAndDeleteIndex" \
  -Dtests.seed=A43FDB00C4693097 \
  -Dtests.security.manager=true \
  -Dtests.locale=pt-BR \
  -Dtests.timezone=America/Argentina/San_Luis \
  -Druntime.java=11

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.snapshots.SnapshotResiliencyTests.testConcurrentSnapshotDeleteAndDeleteIndex" \
  -Dtests.seed=A43FDB00C4693097 \
  -Dtests.security.manager=true \
  -Dtests.locale=pt-BR \
  -Dtests.timezone=America/Argentina/San_Luis \
  -Druntime.java=11
```

**Reproduces locally?**:
No

**Applicable branches**:
master

**Failure history**:
Last failure from over a month ago so probably irrelevant.


**Failure excerpt**:

```
java.lang.AssertionError: expected:<SUCCESS> but was:<PARTIAL>
	at __randomizedtesting.SeedInfo.seed([A43FDB00C4693097:F1742EA707B4B2DA]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.snapshots.SnapshotResiliencyTests.testConcurrentSnapshotDeleteAndDeleteIndex(SnapshotResiliencyTests.java:779)
```

--- a/server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java
+++ b/server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java
@@ -776,8 +776,8 @@ public class SnapshotResiliencyTests extends ESTestCase {
         assertThat(snapshotIds, hasSize(1));
 
         final SnapshotInfo snapshotInfo = repository.getSnapshotInfo(snapshotIds.iterator().next());
-        assertEquals(SnapshotState.SUCCESS, snapshotInfo.state());
         if (partialSnapshot) {
+            assertThat(snapshotInfo.state(), either(is(SnapshotState.SUCCESS)).or(is(SnapshotState.PARTIAL)));
             // Single shard for each index so we either get all indices or all except for the deleted index
             assertThat(snapshotInfo.successfulShards(), either(is(indices + 1)).or(is(indices)));
             if (snapshotInfo.successfulShards() == indices + 1) {
@@ -787,6 +787,7 @@ public class SnapshotResiliencyTests extends ESTestCase {
                 assertEquals(indexMetadata.getIndex(), firstIndex.get());
             }
         } else {
+            assertEquals(snapshotInfo.state(), SnapshotState.SUCCESS);
             // Index delete must be blocked for non-partial snapshots and we get a snapshot for every index
             assertEquals(snapshotInfo.successfulShards(), indices + 1);
         }

